.github/workflows/event-merge-to-queue.yml

```diff
@@ -31,28 +31,83 @@ jobs:
     uses: ./.github/workflows/task-lint.yml
     secrets: inherit
 
-  test-linux:
-    needs: get-latest-redis-tag
-    uses: ./.github/workflows/flow-linux-platforms.yml
+  # Run tests on multiple platforms - matrix defined directly in event
+  unified-test-matrix:
+    name: ${{ matrix.platform }} (${{ matrix.architecture }})
+    needs: [get-latest-redis-tag]
+    strategy:
+      fail-fast: true  # This will cancel ALL matrix jobs on first failure
+      matrix:
+        include:
+          # macOS platforms
+          - platform: macos
+            architecture: x86_64
+          - platform: macos
+            architecture: aarch64
+          # Linux platforms - x86_64
+          - platform: ubuntu:noble
+            architecture: x86_64
+          - platform: ubuntu:jammy
+            architecture: x86_64
+          - platform: ubuntu:focal
+            architecture: x86_64
+          - platform: rockylinux:8
+            architecture: x86_64
+          - platform: rockylinux:9
+            architecture: x86_64
+          - platform: amazonlinux:2
+            architecture: x86_64
+          - platform: amazonlinux:2023
+            architecture: x86_64
+          - platform: debian:bullseye
+            architecture: x86_64
+          - platform: debian:bookworm
+            architecture: x86_64
+          - platform: mariner:2
+            architecture: x86_64
+          - platform: azurelinux:3
+            architecture: x86_64
+          - platform: alpine:3
+            architecture: x86_64
+          # Linux platforms - ARM64
+          - platform: ubuntu:noble
+            architecture: aarch64
+          - platform: ubuntu:jammy
+            architecture: aarch64
+          - platform: ubuntu:focal
+            architecture: aarch64
+          - platform: rockylinux:8
+            architecture: aarch64
+          - platform: rockylinux:9
+            architecture: aarch64
+          - platform: amazonlinux:2023
+            architecture: aarch64
+          - platform: debian:bullseye
+            architecture: aarch64
+          - platform: debian:bookworm
+            architecture: aarch64
+          - platform: azurelinux:3
+            architecture: aarch64
+          - platform: alpine:3
+            architecture: aarch64
```

Comment:
Can't we generate this matrix (all_os X all_arch) style?
I don't like that we don't have these list in a single place. It's going to be hard to maintain